### PR TITLE
Prepare v1.7.0-rc3

### PR DESCRIPTION
**Pull request by jrick**
Updates dcrwallet to release-v1.7.0-rc3 tag.

Updates dcrdex and dcrdex-assets to to v0.4.0-rc3 tags.

### CODE REVIEW

**Comment by jrick** [comment]
```
2022/01/18 14:54:18 dist "dcrinstall-manifests" manifest hash: SHA256:1c4978e3bba4adf16a99ee0e1ce32fdaf4228a306acd1bfb9d4313efab3ffca9 (go version go1.17.6 openbsd/amd64)
```

**Review by davecgh — COMMENTED** [review]
```
windows: dist "dcrinstall-manifests" manifest hash: SHA256:1c4978e3bba4adf16a99ee0e1ce32fdaf4228a306acd1bfb9d4313efab3ffca9 (go version go1.17.6 windows/amd64)
linux: dist "dcrinstall-manifests" manifest hash: SHA256:1c4978e3bba4adf16a99ee0e1ce32fdaf4228a306acd1bfb9d4313efab3ffca9 (go version go1.17.6 linux/amd64)
```

**Review by alexlyp — APPROVED** [review]

**Review by chappjc — APPROVED** [review]
2022/01/18 14:20:08 dist "dcrinstall-manifests" manifest hash: SHA256:1c4978e3bba4adf16a99ee0e1ce32fdaf4228a306acd1bfb9d4313efab3ffca9 (go version go1.17.6 linux/amd64)

File: main.go [hunk 1]
```diff
@@ -78,10 +78,10 @@ type manifestLine struct {
 type manifest []manifestLine
 
 const (
-	decredRelver = "v1.7.0-rc2"
-	dexcRelver   = "v0.4.0-rc2"
-	ldVersion    = "1.7.0-rc2"
-	prerelease   = "rc2"
+	decredRelver = "v1.7.0-rc3"
+	dexcRelver   = "v0.4.0-rc3"
+	ldVersion    = "1.7.0-rc3"
```
Comment (chappjc) [inline main.go:83]: When dex decides to go 1.x (when the APIs and comms protocols stabilize), we can jump straight to 1.7.x or whatever it happens to coincide with.

Comment (jrick) [reply main.go:83]: noted, thx

**Pull request merged**